### ISSUE TITLE
Confusing variable shadowing

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#6620]
This program does not recurse, should this not be a warning? 

![image](https://github.com/flix/flix/assets/32037926/656c1504-2026-4cea-a31f-78b8c1643731)

**Comment by magnus-madsen** [comment]
Does the `use` shadow the call to `example1` or what is going?

**Comment by JonathanStarup** [comment]
Yeah, and p calls the external `example1` function instead of a recursive call, which seems like a weird ordering if it should be allowed at all
